client/my-sites/themes/theme-options.js

```diff
@@ -203,6 +203,41 @@ function getAllThemeOptions( { translate, isFSEActive } ) {
 			isPremiumThemeAvailable( state, themeId, siteId ),
 	};
 
+	// WPCOM-specific plan upgrade for community themes.
+	const upgradePlanForDotOrgThemes = {
+		label: translate( 'Upgrade to activate', {
+			comment: 'label prompting user to upgrade the WordPress.com plan to activate a certain theme',
+		} ),
+		extendedLabel: translate( 'Upgrade to activate', {
+			comment: 'label prompting user to upgrade the WordPress.com plan to activate a certain theme',
+		} ),
+		header: translate( 'Upgrade on:', {
+			context: 'verb',
+			comment: 'label for selecting a site for which to upgrade a plan',
+		} ),
+		getUrl: ( state, themeId, siteId ) => {
+			const { origin = 'https://wordpress.com' } =
+				typeof window !== 'undefined' ? window.location : {};
+			const slug = getSiteSlug( state, siteId );
+
+			const redirectTo = encodeURIComponent(
+				`${ origin }/marketplace/theme/${ themeId }/install/${ slug }`
```

Comment:
This redirect_to is taken from the old CTA removed in https://github.com/Automattic/wp-calypso/pull/94264/files.